### PR TITLE
Catalog of my things 

### PR DESCRIPTION
**Pull request by ibugithub**
-Insert and query data in SQL
-Use primary key & foreign key mechanisms for joining tables.
-Understand the different types of relationships between tables.
-Query multiple tables.
-Prepare complex queries that answer analytical questions.
-Run a program using the command line.
-Use Ruby syntax for basic programming operations.
-Apply Ruby best practices and language style guides in code.
-Describe the SOLID principles of OOP.
-Implement classes and objects in Ruby.
-Understand the four main principles of OOP.
-Implement encapsulation and inheritance with Ruby.
-Create a UML class diagram.
-Explain the difference between associations, aggregations, and composition in OOP.
-Set up associations between classes and objects.
-Store data in files.
-Build interactive console apps.
-Recognize the value of making equal contributions to group projects to produce the best outcome.

video:
https://drive.google.com/file/d/11Sc9g1w5ReIRNrI9j4NfRqMC0LcW5bOS/view?usp=sharing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,152 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/ibugithub/List-my-catalog.git).
```
Comment (FrederickMih) [inline README.md]: - [ ] Let's kindly add the correct issue page

File: genre.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Genre
+  attr_accessor :items, :name, :id
+
+  def initialize(name)
+    @id = rand(1..1000)
+    @name = name
+    @items = []
+  end
+
+  def add_item(item)
```
Comment (FrederickMih) [inline genre.rb:10]: - [ ] Let's kindly pass genre itself to item as you have done with lable. Same to author.

`item.genre = self`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# puts 'author last name?'
+# author = gets.chomp
+# puts 'genre?'
+# genre = gets.chomp
+# puts 'source?'
+# source = gets.chomp
+# puts 'label?'
+# label = gets.chomp
+# puts 'author last name?'
+# author = gets.chomp
```
Comment (FrederickMih) [inline book.rb]: - [ ] Let's make sure our codebase is clean by removing commented codes.

File: music_album.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require './item'
+
+class MusicAlbum < Item
+  attr_accessor :on_spotify
+
+  def initialize(pub_date, on_spotify, archived: false)
+    super(pub_date, archived: archived)
+    @on_spotify = on_spotify
+  end
```
Comment (FrederickMih) [inline music_album.rb]: - [ ] I think the `can_be_archived?` method name should remain the same as in other classes. This method shouldn't be structure inside the constructor. The method return true if parent's method returns true **AND** if on_spotify equals true. **Hints**

```ruby
 def can_be_archived?
      super && @on_spotify == true
 end
```

File: game.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require './item'
+
+class Game < Item
+  attr_accessor :pub_date, :multiplayer, :last_played
+
+  def initialize(pub_date, multiplayer, last_played)
+    super(pub_date)
+    @multiplayer = multiplayer
+    @last_played = last_played
+  end
```
Comment (FrederickMih) [inline game.rb:10]: - [ ] kindly refer to the comment above :up: We just needs to override the behavior of the parent(super) class hence the method name should be the same

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @ibugithub, @benja27  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Highlights :green_circle: 

- Data is correctly save after quitting the app :+1: 
- Foreign keys were set up properly in each table :+1: 

## Required Changes ♻️

- [ ] **Video Presentation:** Great job for taking your time to present all the features of your project :+1: unfortunately the presentation exceed 5 minutes. We are always expected to adhere to the requirements of our client. Hence I'll kindly suggest to record another video and be brief with the presentation. Again, the presentation was not really professional because of the presence of noise in the background. 

_Check the comments under the review._

## Optional suggestions
**N/A**
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: game.rb (hunk 5 above)
Comment (ibugithub) [reply game.rb:10]: Solved them and Thanks for mentioning them.

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**